python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -134,7 +134,10 @@ def _as_config_dict(self) -> Mapping[str, Any]:
             field = self.__fields__.get(key)
             if field and value is None and not _is_pydantic_field_required(field):
                 continue
-            output[key] = value
+            if field:
+                output[field.alias] = value
```

Comment:
if the config doesn't have an alias, the `alias` attribute is just the name of the config